instrumentation/src/main/java/io/opentelemetry/android/internal/services/AppInfoService.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.internal.services;
+
+import android.content.Context;
+import android.os.Build;
+import android.os.storage.StorageManager;
+import androidx.annotation.RequiresApi;
+import androidx.annotation.WorkerThread;
+import java.io.File;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Utility to get information about the host app.
+ *
+ * <p>This class is internal and not for public use. Its APIs are unstable and can change at any
+ * time.
+ */
+public class AppInfoService implements Service {
```

Comment:
Looking at the public surface/outline of this class, it doesn't _really_ seem to deal with `AppInfo` very much....it deals more with the cache. Can we call it `CacheStorageService` instead?

Reply:
Sure, makes sense. It's updated now.